src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -56,9 +56,56 @@ def get_preference(
         information  # type: Sequence[Tuple[Requirement, Candidate]]
     ):
         # type: (...) -> Any
+        """Produce a sort key for given requirement based on preference.
+
+        The lower the return value is, the more preferred this group of
+        arguments is.
+
+        Currently pip considers the followings in order:
+
+        * Prefer if any of the known requirements points to an explicit URL.
+        * If equal, prefer if any requirements contain `===` and `==`.
```

Comment:
Update the comment to reflect recent changes?
```suggestion
        * If equal, prefer if any requirements contain `===` and `==`.
        * If equal, prefer if requirements include version constraints (e.g., `>=` or `<`).
```